### PR TITLE
Add generation script

### PR DESCRIPTION
**Pull request by thomasw21**
Modified script from @younesbelkada version.

Essentially what it does is create an interactive way of asking the model to generate. Allows for local prompting beyond what's proposed via the inference API. 

cc @TevenLeScao @SaulLu @stas00

### CODE REVIEW

File: evaluation/generation/generate.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+def generate_from_text(model, text, tokenizer, max_length=200, greedy=False):
+    input_ids = tokenizer.encode(text, return_tensors='pt').to("cuda:0")
+    max_length = input_ids.size(-1) + max_length
+    if greedy:
+        greedy_output = model.generate(input_ids.to('cuda:0'), max_length=max_length)
+    else:
+        greedy_output = model.generate(input_ids.to('cuda:0'), max_length=max_length, do_sample=True, top_k=0)
+    return {
+        "inputs": text,
+        "outputs": tokenizer.decode(greedy_output[0], skip_special_tokens=True)
```
Comment (thomasw21) [inline evaluation/generation/generate.py:29]: Maybe we can see if we can get only the "output" in the sense that current version will also contain the prompt.

**Comment by stas00** [comment]
Please tell me again why do we want an extremely slow naive parallelization when we have deepspeed that can process 8 text streams at the same time and there is no need to write any new scripts?

https://huggingface.co/docs/transformers/main/main_classes/deepspeed#custom-deepspeed-zero-inference

**Comment by thomasw21** [comment]
The idea is to have an interactive script. You enter text, it generates, so only one text stream. It was really for me find good generations.

**Review by SaulLu — APPROVED** [review]
Thank for this handy script!

**Comment by younesbelkada** [comment]
I made a PR here that is compatible with the latest release of transformers: https://github.com/bigscience-workshop/bigscience/pull/43

**Comment by thomasw21** [comment]
Closing in favor of #43

**Pull request closed without merging**